include/fmt/compile.h

```diff
@@ -789,7 +794,17 @@ FMT_INLINE std::basic_string<typename S::char_type> format(const S&,
   }
 #endif
   constexpr auto compiled = detail::compile<Args...>(S());
+#ifdef __cpp_if_constexpr
+  if constexpr (std::is_same<remove_cvref_t<decltype(compiled)>,
```

Comment:
This doesn't need to be a constexpr if. The compiler will optimize the unneeded branch away.

Reply:
The problem without using `constexpr if` is that else branch becomes invalid (before the optimization), for the same reason the current implementation fails to fall back to runtime API: https://github.com/alexezeder/fmt/runs/2007151954?check_suite_focus=true#step:5:70, https://github.com/alexezeder/fmt/runs/2007151989?check_suite_focus=true#step:5:70.
So it has to be `constexpr if` to omit else branch completely.
Of course, another option is to fix `char_t` for `unknown_format` (and maybe more fixes), but it seems unnecessary with `constexpr if`.